### PR TITLE
Add option to output unsorted schema

### PR DESCRIPTION
**Pull request by vigneshmanick**
## Change Summary

Add toggle so that the schema output `model_dump`  can be sorted or unsorted. The sorting was added in https://github.com/pydantic/pydantic/pull/6043. 

## Related issue number

#7580

## Checklist

* [x] The pull request title is a good summary of the changes - it will be used in the changelog
* [x] Unit tests for the changes exist
* [x] Tests pass on CI
* [ ] Documentation reflects the changes where applicable
* [x] My PR is ready to review, **please add a comment including the phrase "please review" to assign reviewers**

Selected Reviewer: @davidhewitt

### CODE REVIEW

**Comment by vigneshmanick** [comment]
@sydney-runkle  have tried to create an option for unsorted output.  Let me know if this is ok or totally off the mark :)

Please review

File: docs/concepts/json_schema.md [hunk 1]
```diff
@@ -748,8 +748,10 @@ from pydantic.json_schema import GenerateJsonSchema
 
 
 class MyGenerateJsonSchema(GenerateJsonSchema):
-    def generate(self, schema, mode='validation'):
-        json_schema = super().generate(schema, mode=mode)
+    def generate(self, schema, mode='validation', sort_schema=True):
+        json_schema = super().generate(
+            schema, mode=mode, sort_schema=sort_schema
+        )
```
Comment (sydney-runkle) [inline docs/concepts/json_schema.md:754]: Thanks for making sure to update the example in the docs 😄

File: pydantic/json_schema.py [hunk 2]
```diff
@@ -436,7 +439,10 @@ def generate(self, schema: CoreSchema, mode: JsonSchemaMode = 'validation') -> J
         # json_schema['$schema'] = self.schema_dialect
 
         self._used = True
-        return _sort_json_schema(json_schema)
+        if sort_schema:
+            return _sort_json_schema(json_schema)
+        else:
+            return json_schema
```
Comment (sydney-runkle) [inline pydantic/json_schema.py:445]:
```suggestion
        return json_schema
```

File: tests/test_json_schema.py [hunk 3]
```diff
@@ -4264,8 +4264,8 @@ class Foobar(BaseModel):
 
 def test_override_generate_json_schema():
     class MyGenerateJsonSchema(GenerateJsonSchema):
-        def generate(self, schema, mode='validation'):
-            json_schema = super().generate(schema, mode=mode)
+        def generate(self, schema, mode='validation', sort_schema=True):
+            json_schema = super().generate(schema, mode=mode, sort_schema=sort_schema)
```
Comment (sydney-runkle) [inline tests/test_json_schema.py:4268]: Could you add a section to this test that verifies the operation of the `sort_schema=False` flag in the case of a custom generate json schema function?

**Review by sydney-runkle — CHANGES_REQUESTED** [review]
Looks great overall - just a few change requests.

Thanks so much for your contribution - I'm sure this feature will be highly appreciated 😄.

**Comment by sydney-runkle** [comment]
@adriangb, could you take a look at this PR? Others on the team mentioned that you've contributed significantly to the sorting logic, so would love to get your input 😄.

File: tests/test_json_schema.py (hunk 3 above)
Comment (vigneshmanick) [reply tests/test_json_schema.py:4268]: do you mean parameterize the flag?

**Comment by sydney-runkle** [comment]
Closing in favor of #7817

File: tests/test_json_schema.py (hunk 3 above)
Comment (wheelly) [reply tests/test_json_schema.py:4268]: Why was it closed? It really a very nice thing to have. Why would we need to sort fields in schema examples like {user..., password...} into {password..., user...}  - it looks awful. Pls reopen it guys

Comment (sydney-runkle) [reply tests/test_json_schema.py:4268]: See https://docs.pydantic.dev/latest/concepts/json_schema/#json-schema-sorting

**Pull request closed without merging**